resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -155,7 +155,7 @@ private[spark] class KubernetesClusterSchedulerBackend(
   private def allocateNewExecutorPod(): (String, Pod) = {
     val executorKubernetesId = UUID.randomUUID().toString.replaceAll("-", "")
     val executorId = EXECUTOR_ID_COUNTER.incrementAndGet().toString
-    val name = s"exec$executorKubernetesId"
+    val name = s"$kubernetesDriverServiceName-exec$executorKubernetesId"
```

Comment:
I ran into name length constraints before while trying to figure out the name. But if this works then I'm fine with this change.